adb_graphics/datahandler/gribdata.py

```diff
@@ -112,17 +112,24 @@ def field_diff(self, values, variable2, level2, **kwargs):
 
         return diff
 
-    def field_mean(self, values, variable, levels, **kwargs):
+    def field_mean(self, values, variable, levels, global_levels, **kwargs):
 
         # pylint: disable=unused-argument
 
         ''' Returns the mean of the values. '''
 
         fsum = np.zeros_like(values)
-        for level in levels:
-            val_lev = self.values(name=variable, level=level)
-            fsum = fsum + val_lev
-            val_lev.close()
+
+        if 'global' in self.model:
+            for level in global_levels:
+                val_lev = self.values(name=variable, level=level)
+                fsum = fsum + val_lev
+                val_lev.close()
+        else:
+            for level in levels:
+                val_lev = self.values(name=variable, level=level)
+                fsum = fsum + val_lev
+                val_lev.close()
```

Comment:
So we don't have to duplicate the algorithm here, it would be cleaner to choose the levels you want to plot with logic, then do the heavy lifting using that variable. Something like this:

```
chosen_levels = global_levels if 'global' in self.model else levels
for level in chosen_levels:
    val_lev = ...
    fsum += val_lev
    val_lev.close()
```

Reply:
Thanks, I suspected there was a better way to do this.